config/v3_0/types/node_test.go

```diff
@@ -20,79 +20,82 @@ import (
 
 	"github.com/coreos/ignition/v2/config/shared/errors"
 	"github.com/coreos/ignition/v2/config/util"
-	"github.com/coreos/ignition/v2/config/validate/report"
+
+	"github.com/ajeddeloh/vcontext/path"
+	"github.com/ajeddeloh/vcontext/report"
 )
 
 func TestNodeValidatePath(t *testing.T) {
 	node := Node{Path: "not/absolute"}
-	rep := report.ReportFromError(errors.ErrPathRelative, report.EntryError)
-	if receivedRep := node.ValidatePath(); !reflect.DeepEqual(rep, receivedRep) {
+	rep := report.Report{}
+	rep.AddOnError(path.ContextPath{}.Append("path"), errors.ErrPathRelative)
+	if receivedRep := node.Validate(path.ContextPath{}); !reflect.DeepEqual(rep, receivedRep) {
 		t.Errorf("bad error: want %v, got %v", rep, receivedRep)
 	}
 }
 
-func intToPtr(x int) *int {
-	return &x
-}
-
 func TestNodeValidateUser(t *testing.T) {
 	tests := []struct {
 		in  NodeUser
-		out report.Report
+		out error
 	}{
 		{
-			in:  NodeUser{intToPtr(0), util.StrToPtr("")},
-			out: report.Report{},
+			in:  NodeUser{util.IntToPtr(0), util.StrToPtr("")},
```

Comment:
Any specific reason for using named fields here in test data? I see  in rest of diff that's not the case.

Reply:
Nope! Old habits die hard. Will update!